### PR TITLE
[HWORKS-936] Explicit model provenance

### PR DESCRIPTION
**Pull request by o-alex**
This PR adds/fixes/changes...
- please summarize your changes to the code 
- and make sure to include all changes to user-facing APIs

JIRA Issue: -

Priority for Review: -

Related PRs: -

**How Has This Been Tested?**

- [ ] Unit Tests
- [ ] Integration Tests
- [ ] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

**Review by bubriks — CHANGES_REQUESTED** [review]

File: python/hsfs/core/feature_view_api.py [hunk 1]
```diff
@@ -289,3 +289,37 @@ def get_parent_feature_groups(self, name, version):
...
+
+    def get_generated_models(self, name, version):
+        """Get the generated models using this feature view, based on explicit
+        provenance. These models can be accessible or inaccessible. Explicit
+        provenance does not track deleted generated model links, so deleted
+        will always be empty.
+        For inaccessible models, only a minimal information is returned.
+
+        # Arguments
+            feature_view_instance: Metadata object of feature view.
```
[the remark this answers is not in the document]
Comment (o-alex) [reply python/hsfs/core/feature_view_api.py:301]: copy paste from hsml. sorry. fixed

File: python/hsfs/core/explicit_provenance.py [hunk 2]
```diff
@@ -103,6 +126,17 @@ def from_response_json(json_dict: dict):
                 Artifact.MetaType.INACCESSIBLE,
                 link_json["artifact"]["href"],
             )
+        elif bool(link_json["accessible"]):
+            return Artifact(
+                link_json["artifact"]["featurestore_name"],
```
[the remark this answers is not in the document]
Comment (o-alex) [reply python/hsfs/core/explicit_provenance.py:131]: right about project. and simplified the if else.

File: python/hsfs/core/feature_view_engine.py [hunk 3]
```diff
@@ -783,6 +783,23 @@ def get_parent_feature_groups(self, feature_view_obj):
...
 
+    def get_generated_models(self, feature_view_obj):
+        """Get the generated models using this feature view, based on explicit
+        provenance. These models can be accessible or inaccessible. Explicit
+        provenance does not track deleted generated model links, so deleted
+        will always be empty.
+        For inaccessible models, only a minimal information is returned.
+
+        # Arguments
+            feature_view_instance: Metadata object of feature view.
```
[the remark this answers is not in the document]
Comment (o-alex) [reply python/hsfs/core/feature_view_engine.py:794]: fixed

File: python/hsfs/core/feature_view_api.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bubriks) [reply python/hsfs/core/feature_view_api.py:301]: fix was not pushed

**Comment by bubriks** [comment]
use this PR instead: https://github.com/logicalclocks/feature-store-api/pull/1254

**Pull request closed without merging**